client/pdo/client/scripts/eservicedatabaseCLI.py

```diff
@@ -0,0 +1,282 @@
+    'etc'  : ContractEtc,
+    'home' : ContractHome,
+    'host' : ContractHost,
+    'keys' : ContractKeys,
+    'logs' : ContractLogs,
+    'ledger' : LedgerURL
+}
+
+# -----------------------------------------------------------------
+# -----------------------------------------------------------------
+def Main(commands) :
+    # parse out the configuration file first
+    conffiles = [ 'pcontract.toml' ]
+    confpaths = [ ".", "./etc", ContractEtc ]
+
+    parser = argparse.ArgumentParser()
+    
+    parser.add_argument('--config', help='configuration file', nargs = '+')
+    parser.add_argument('--config-dir', help='configuration file', nargs = '+')
+
+    parser.add_argument('--eservice-url', help='service urls',  nargs='+')
+    parser.add_argument('--eservice-name', help='service names',  nargs='+')
+    parser.add_argument('--eservice-db', help='json file for database', type=str)
+    parser.add_argument('--loglevel', help='Set the logging level', default='INFO')
+    parser.add_argument('--logfile', help='Name of the log file', default='__screen__')
+
+    options = parser.parse_args()
+
+    # first process the options necessary to load the default configuration
+    if options.config :
+        conffiles = options.config
+
+    if options.config_dir :
+        confpaths = options.config_dir
+
+    global config_map
+    
+    try :
+        config = pconfig.parse_configuration_files(conffiles, confpaths, config_map)
+    except pconfig.ConfigurationException as e :
+        logger.error(str(e))
+        sys.exit(-1)
+
+    # set up the logging configuration
+    if config.get('Logging') is None :
+        config['Logging'] = {
+            'LogFile' : '__screen__',
+            'LogLevel' : 'INFO'
+        }
+    if options.logfile :
+        config['Logging']['LogFile'] = options.logfile
+    if options.loglevel :
+        config['Logging']['LogLevel'] = options.loglevel.upper()
+
+    plogger.setup_loggers(config.get('Logging', {}))
+
+    # process the reset of the command parameters
+
+    # set up the service configuration
+    if config.get('Service') is None :
+        config['Service'] = {
+            'EnclaveServiceURLs' : [],
+            'EnclaveServiceNames' : [],
+            'EnclaveServiceDatabaseFile' : None
+        }
+
+    if options.eservice_url :
+        config['Service']['EnclaveServiceURLs'] = options.eservice_url
+    
+    if options.eservice_name :
+        config['Service']['EnclaveServiceNames'] = options.eservice_name
+
+    if options.eservice_db:
+        config['Service']['EnclaveServiceDatabaseFile'] = options.eservice_db
+    
+   
+    # GO!!!
+    LocalMain(commands, config)
+
+## -----------------------------------------------------------------
+## Entry points
+## -----------------------------------------------------------------
+def Add() :
+    Main(['add'])
+
+def Remove() :
+    Main(['remove'])
+
+def Create() :
+    Main(['create'])
+
+def Update():
+    Main(['update'])    
```

Comment:
any reason why we really need multiple entry points?

Reply:
the  goal was to allow modifcation to a previously available db. the db need not be erased and recreated every time we invoke run-tests. It can be updated with the new enclave_ids for the same names, and urls. the update command allows us to do this. (Of course, i havent implemented this in run-tests, but it is a two line change). 

also, the goal was to work with a db that a client might get from another client, and modify before usage.